### PR TITLE
Async with multiple threads

### PR DESCRIPTION
**Pull request by nnhw**
Attempt to build working data exchange with multiple threads

### CODE REVIEW

File: impl/include/link_transceiver.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    void start() override;
+    void stop() override;
+
+    void send(const QByteArray& data);
+
+    //protected:
+    //    void timerEvent(QTimerEvent* event) override;
+
+signals:
+    void receivedData(const QByteArray& data);
```
Comment (MishkaRogachev) [inline impl/include/link_transceiver.h]: Сonst refs are dangerous in signals - `void receivedData(QByteArray data);`

File: impl/include/mavlink_transceiver.h [hunk 2]
```diff
@@ -29,14 +30,19 @@ public slots:
...
 
 private slots:
-    void receiveData();
-    void parseMessage(const QByteArray& data);
-    void send(const mavlink_message_t& message);
+    void receiveData(const QByteArray& data);
+
+    void sendMessage(const mavlink_message_t& message);
+
+signals:
```
Comment (MishkaRogachev) [inline impl/include/mavlink_transceiver.h]: signals before private slots

File: impl/include/link_transceiver.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+private:
+    void receiveData();
+    data_source::LinkPtr const m_link;
```
Comment (MishkaRogachev) [inline impl/include/link_transceiver.h:35]: empty line between methods and members

File: impl/include/mavlink_transceiver.h [hunk 4]
```diff
@@ -29,14 +30,19 @@ public slots:
...
+    void sendData(const QByteArray& data);
 
 private:
+    void parseMessage(const QByteArray& data);
```
Comment (MishkaRogachev) [inline impl/include/mavlink_transceiver.h:37]: empty line between methods and members

File: impl/include/mavlink_transceiver.h [hunk 5]
```diff
@@ -29,14 +30,19 @@ public slots:
...
     int m_timerId = 0;
     MavlinkHandlerContext m_context;
     data_source::LinkPtrMap const m_links;
+    QVector<data_source::LinkTransceiverThreaded*> m_linkTransceiverThreaded;
```
Comment (MishkaRogachev) [inline impl/include/mavlink_transceiver.h]: Can we use here interface class?
`QVector<data_source::ILinkTransceiver> m_linkTransceivers`

**Review by MishkaRogachev — CHANGES_REQUESTED** [review]
Some CS questions

File: impl/include/mavlink_transceiver.h (hunk 5 above)
Comment (nnhw) [reply impl/include/mavlink_transceiver.h]: yes, done

File: impl/include/link_transceiver.h (hunk 1 above)
Comment (nnhw) [reply impl/include/link_transceiver.h]: changed for by value passing

**Review by MishkaRogachev — APPROVED** [review]

**Pull request merged**